tools/c7n_azure/c7n_azure/resources/event_subscription.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from c7n_azure.provider import resources
+from c7n_azure.query import QueryResourceManager
+from c7n_azure.utils import ThreadHelper
+
+from c7n.actions import BaseAction
+from c7n.filters.core import type_schema
+
+
+@resources.register('eventsubscription')
+class EventSubscription(QueryResourceManager):
+
+    class resource_type(object):
+        service = 'azure.mgmt.eventgrid'
+        client = 'EventGridManagementClient'
+        enum_spec = ('event_subscriptions', 'list_global_by_subscription', None)
+        default_report_fields = (
+            'name',
+            'properties.destination.endpointType',
+            'properties.topic'
+        )
+
+
+@EventSubscription.action_registry.register('delete')
+class Delete(BaseAction):
+    schema = type_schema('delete')
+
+    def process(self, resources, event=None):
+        return ThreadHelper.execute_in_parallel(resources=resources,
```

Comment:
probably out of scope for this PR, but we can move this execute_in_parallel to some AzureBaseAction, so all Azure actions have it

Reply:
Good idea! I'll follow-up with a PR that does that and refactors all the other parallel calls we make (or should be making).